src/harness/util.py

```diff
@@ -408,3 +408,50 @@ def filterChannelsByFrequencyRange(channels, freq_range):
       and
       channel['frequencyRange']['highFrequency'] <= freq_range['highFrequency']
   ]
+
+def _orderAttributes(obj):
+    if isinstance(obj, dict):
+        return sorted((k, _orderAttributes(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return sorted(_orderAttributes(x) for x in obj)
+    else:
+        return obj
+
+def compareDictWithUnorderedLists(first_dict, second_dict):
+  """ Deep comparison of two dictionaries
+
+  Args:
+    first_dict: first dictionary to be compared.
+    second_dict: second dictionary to be compared.
+ Returns: boolean set to true if the dictionaries are equal and false otherwise.
+  """
+  return _orderAttributes(first_dict) == _orderAttributes(second_dict)
+
+def areTwoPpasEqual(first_ppa, second_ppa, delta=10):
+  """ Deep comparison of two PPAs considering
+
+  Args:
+    first_ppa: a dictionary contains a reference of PPA data to compare to,
```

Comment:
Is it fair to describe this better as:
```
first_ppa: a dict with key 'zone' holding a GeoJSON feature collection with first feature
     defining the PPA zone as a polygon (or multipolygon) geometry.
```
By the way,  you are deep-comparing the objects. Is your intent to compare only the element with key 'zone' ?
In that case it would be better to have this utility library taking this element, which is basically a GeoJson feature collection.
Also is it expected that we always use a FeatureCollection here (and not a Feature) ?

Reply:
the comment is not valid anymore, and I need to deep comparing all the PPA parameters, but treating the polygons differently.
and according to the spec the zone is always FeatureCollection.